contracts/src/v0.8/llo-feeds/v0.4.0/DestinationVerifier.sol

```diff
@@ -323,19 +323,20 @@ contract DestinationVerifier is IDestinationVerifier, ConfirmedOwner, TypeAndVer
         return timestamp;
     }
 
-    function findActiveConfig(uint256 timestamp) internal view returns (DONConfig memory){
+   function findActiveConfig(uint256 timestamp) internal view returns (DONConfig memory){
         DONConfig memory activeDONConfig;
 
         // 99% of the time the signer config will be the last index, however for historic reports generated by a previous configuration we'll need to cycle back
-        for(uint256 i = s_DONConfigs.length - 1; i >= 0; --i) {
-            if(s_DONConfigs[i].activationTime < timestamp) {
-                activeDONConfig = s_DONConfigs[i];
+        for(uint256 i = s_DONConfigs.length; i > 0; i--) {
```

Comment:
@Fletch153 

this for loop could make `i` take negative values, which results in an underflow error.

```solidity
for(uint256 i = s_DONConfigs.length - 1; i >= 0; --i) {
```

I think this might be read easier as a `while` loop, but I wonder how that goes gas-wise 🤔

Reply:
While loop could work, if we can avoid decrementing the index each time in the loop that would save some gas.

Reply:
Changing it to a signed int256 should fix this problem I think, that saves us decrementing it each loop using extra gas

Reply:
Fell free to update it as you see fit. For loops and while loops are equally efficient, it's more about computational and memory costs. Right now we're decrementing I twice each loop